src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,26 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+
+    private static final int MIN_AGE = 35;
+    private static final int MIN_PERIOD = 10;
+    private static final String UKRAINIAN = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (candidate.getAge() < MIN_AGE) {
```

Comment:
create separate method `checkTimeLivingInCountry(candidate);`

```suggestion
        return candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(REQUIRED_NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```

Reply:
Changes have been made